clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3576,14 +3576,25 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
 
   if (UseSYCLTriple) {
     // We want to compile sycl kernels.
-    if (types::isCXX(Input.getType()))
-      CmdArgs.push_back("-std=c++11");
     CmdArgs.push_back("-fsycl-is-device");
     // Pass the triple of host when doing SYCL
-    std::string NormalizedTriple =
-        llvm::Triple(llvm::sys::getProcessTriple()).normalize();
+    auto AuxT = llvm::Triple(llvm::sys::getProcessTriple());
+    std::string NormalizedTriple = AuxT.normalize();
     CmdArgs.push_back("-aux-triple");
     CmdArgs.push_back(Args.MakeArgString(NormalizedTriple));
+
+    bool IsMSVC = AuxT.isWindowsMSVCEnvironment();
+    if (types::isCXX(Input.getType()))
+      CmdArgs.push_back(IsMSVC ? "-std=c++14" : "-std=c++11");
+    if (IsMSVC) {
+      CmdArgs.push_back("-fms-extensions");
+      VersionTuple MSVT = TC.computeMSVCVersion(&D, Args);
+      if (!MSVT.empty())
+        CmdArgs.push_back(Args.MakeArgString("-fms-compatibility-version=" + MSVT.getAsString()));
+      else
+        CmdArgs.push_back("-fms-compatibility-version=191526730");
```

Comment:
It is a bit uncomfortable to have this hardcoded version here.
Wouldn't it be better to assert that MSVT is not empty?

Reply:
for some reason MSVT is coming back empty, i'll see about debugging that

Reply:
in the mean time can we use this as workaround that needs fixing?

Reply:
Please spend some time debugging.
After that if nothing can be done with that, then it could be some oldest MSVC ver that still usable for the build (I suppose compiler cannot be built with older than MSVC-2017).